File: COMPILADOR/kokoslan_/src/src/java/ast/KoKoNum.java
/**
 @author loriacarlos@gmail.com
 
*/
package src.java.ast;

import src.java.eval.KoKoContext;
import src.java.eval.KoKoNumValue;
import src.java.eval.KoKoValue;

public class KoKoNum extends KoKoAtom<Double>{
   public KoKoNum(Double value){
      super(value);
   }
   @Override
   public KoKoValue eval(KoKoContext ctx){
	   return new KoKoNumValue(getValue());
	   
   }
   
}

File: COMPILADOR/kokoslan/src/KoKoKotlin/ast/KoKoNum.kt
/**
 * @author loriacarlos@gmail.com
 */
package KoKoKotlin.ast

import KoKoKotlin.eval.KoKoContext
import KoKoKotlin.eval.KoKoNumValue
import KoKoKotlin.eval.KoKoValue


class KoKoNum(value: Double) : KoKoKotlin.ast.KoKoAtom<Double>(value) {
    override fun eval(ctx: KoKoContext): KoKoValue {
        return KoKoNumValue(value)

    }

}